.github/workflows/format.yml

```diff
@@ -0,0 +1,26 @@
+name: Clang format
+
+on: [pull_request]
+
+jobs:
+  clang-format-8:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Run clang-foramt on changed files
+        run: |
+          set -x
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+          git fetch origin pull/${{ github.event.pull_request.number }}/head:${{ github.event.pull_request.head.ref }}
+          BASE_COMMIT=$(git rev-parse ${{ github.event.pull_request.base.sha }})
+          COMMIT_FILES=$(git diff --name-only ${BASE_COMMIT} | grep -i -v LinkDef)
```

Comment:
What about CMakeLists, bash scripts and python files. With clang-format behave?

Reply:
correct, good point, I am adding a restriction to *.cxx and *.h only, excluding LinkDef